### PR TITLE
fix: fix nvd3 and histogram regressions and migrate control panels to TS

### PR DESCRIPTION
**Pull request by villebro**
🏆 Enhancements / 🐛 Bug Fix
A recent regression in the histogram viz from #869 highlighted the need to migrate all JavaScript based control panels to TypeScript. The NVD3 timeseries charts also experienced a regression where the advanced analytics section was lost. This PR does the following changes:
- Calendar: Migrate control panel to TS and fix incorrect import
- Chord: Migrate control panel to TS
- Heatmap: Migrate control panel to TS
- Histogram: Migrate control panel to TS and fix incorrect section location
- Horizon: Migrate control panel to TS
- Map Box: Migrate control panel to TS
- NVD3 (Area, Bar, Compare, Line): Move legacy Time section from `NVD3Controls.tsx` to individual chart panels.
- Partition: Migrate control panel and `OptionDescription.jsx`to TS
- Pivot Table: Migrate control panel to TS
- Rose: Migrate control panel to TS
- Sankey Loop: Migrate control panel to TS
- Treemap: Migrate control panel to TS and fix incorrect import
- World Map: Migrate control panel to TS

### CODE REVIEW

File: plugins/legacy-plugin-chart-calendar/src/controlPanel.ts [hunk 1]
```diff
@@ -16,15 +16,16 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+import { t, legacyValidateInteger } from '@superset-ui/core';
 import {
-  t,
-  legacyValidateInteger,
-  D3_TIME_FORMAT_OPTIONS,
```
Comment (villebro) [inline plugins/legacy-plugin-chart-calendar/src/controlPanel.ts:22]: This import was referencing `@superset-ui/core`, but it's actually an export in `@superset/ui/chart-controls`, aka the control didn't work (only showing default value): 
image

File: plugins/legacy-plugin-chart-histogram/src/controlPanel.ts [hunk 2]
```diff
@@ -17,15 +17,20 @@
...
 
-export default {
+const config: ControlPanelConfig = {
   controlPanelSections: [
+    sections.legacyRegularTime,
     {
       label: t('Query'),
       expanded: true,
       controlSetRows: [
-        sections.legacyRegularTime,
```
Comment (villebro) [inline plugins/legacy-plugin-chart-histogram/src/controlPanel.ts:28]: Incorrectly placed section - not picked up due to file originally being `js`.

File: plugins/legacy-plugin-chart-treemap/src/controlPanel.ts [hunk 3]
```diff
@@ -16,10 +16,15 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import { t, D3_FORMAT_OPTIONS, D3_FORMAT_DOCS } from '@superset-ui/core';
```
Comment (villebro) [inline plugins/legacy-plugin-chart-treemap/src/controlPanel.ts:19]: Also imported from wrong package.

**Review by ktmud — APPROVED** [review]
LGTM!

**Pull request merged**